src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -229,13 +229,14 @@ private static int GetJSObjectId(object rawObj)
                 js.GetWrappedObject() ?? h.Target : h.Target;
         }
 
-        private static object BoxInt(int i)
-        {
-            return i;
-        }
-
-        private static object BoxDouble(double d)
+        private static object BoxNumber(double d)
         {
+            if ((d % 1) == 0) {
```

Comment:
I'm fine with Truncate, but now I'm curious how the interpreter handles the two cases.

Reply:
Yeah, interesting point. I imagine Math.Truncate probably performs a fn call, but maybe it's an intrinsic? ```(int)d``` should also work with the caveat that it might do something slightly different for negative numbers, I'd have to check.